WordPress/gutenberg
# Block Supports: Add background image support to Group block

## What?


Fixes #14744, part of #16479. Previous attempt: https://github.com/WordPress/gutenberg/pull/39243

This PR explores adding a Background block support, with a single `backgroundImage` property to begin with. The Group block is opted-in to this support.

Note: this PR originally proposed a Media block support as the container for this sub-feature. The PR currently reflects a Background block support, and the final naming is not (yet) decided — this is currently in discussion in the comments.

## Why?


As discussed in #14744, it is useful for more container blocks to have background image support, rather than just the Cover block. The approach in this PR adds in background image support (within Background) as a generalised block support so that more blocks could use the feature in the future, too. Of note: the approach in this PR dynamically injects the background image and sizing at render time, which gives the block support flexibility, and means we don't need to worry about block deprecations.

## How?


* Add a Background block support and panel to the block inspector
* Add a Background Image control and option for the background block support (based on the Media controls in the Site Logo block)
* Opt the Group block in to using this block support
* Dynamically render an inline style for the background image when an image is set, and default to a background size of `cover` (with the ability for us to customise this value in follow-ups)

## Not covered

* If the approach here becomes viable, then in follow-ups we could add in additional controls and properties for things like:
  * Background position (as in the Cover block)
  * Background repeat
  * Background size for repeating (to control how large or small repeated background images are)

## Testing Instructions





1. In a blocks theme that uses `appearanceTools` (e.g. TT3) add a Group block to a post or page, with a few blocks within it, and maybe some padding.
2. In the block inspector at the right hand side, you should now see a Media panel.
3. Explore adding an image from the media library.
4. Save the post and load the site frontend, and ensure the background image renders.

## To-do

* [x] ~Fix skipSerialization (currently the styles _are_ being serialized when they shouldn't be)~

## Screenshots or screencast 

<img width="1419" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/5e193ef3-6851-4800-a864-aeeaa13b4ec4">

---



The following screengrab demos adding a background image with a transparent background, and then selecting a background image that has no transparency:

https://github.com/WordPress/gutenberg/assets/14988353/82729e70-8a54-427b-b410-033e0a19faa9
